pkg/operation/botanist/secrets.go

```diff
@@ -194,10 +197,43 @@ func (b *Botanist) DeploySecrets(ctx context.Context) error {
 	for name, secret := range secretsManager.DeployedSecrets {
 		b.StoreSecret(name, secret)
 	}
+
 	for _, name := range b.AllSecretKeys() {
 		b.StoreCheckSum(name, utils.ComputeSecretChecksum(b.LoadSecret(name).Data))
 	}
 
+	// copy ssh-keypair.old secret to shoot namespace in seed
+	oldSSHKeyPair, err := infodata.GetInfoData(gardenerResourceDataList, v1beta1constants.SecretNameOldSSHKeyPair)
+	if err != nil {
+		return err
+	}
+	if oldSSHKeyPair != nil {
+		secretConfig := &secrets.RSASecretConfig{
+			Name:       v1beta1constants.SecretNameOldSSHKeyPair,
+			Bits:       4096,
+			UsedForSSH: true,
+		}
+
+		oldSSHKeyPairData, err := secretConfig.GenerateFromInfoData(oldSSHKeyPair)
+		if err != nil {
+			return err
+		}
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      v1beta1constants.SecretNameOldSSHKeyPair,
+				Namespace: b.Shoot.SeedNamespace,
+			},
+		}
+		if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, b.K8sSeedClient.Client(), secret, func() error {
+			secret.Type = corev1.SecretTypeOpaque
+			secret.Data = oldSSHKeyPairData.SecretData()
+			return nil
+		}); err != nil {
+			return err
+		}
```

Comment:
The secret needs to be stored so that it can be used by the operating system config deployment [here](https://github.com/gardener/gardener/blob/4825315b78e020e612958d06fe613518dbf784d2/pkg/operation/botanist/operatingsystemconfig.go#L95):
```
b.StoreSecret(v1beta1constants.SecretNameOldSSHKeyPair, secret)
```
I see that it is currently stored in `SyncShootCredentialsToGarden` but the OSC deployment does not depend on that, so I think there's a chance that `DeployOperatingSystemConfig` can be called in parallel or before `SyncShootCredentialsToGarden`